sympy/stats/crv_types.py

```diff
@@ -2195,6 +2196,110 @@ def Normal(name, mean, std):
             MultivariateNormalDistribution, name, mean, std)
     return rv(name, NormalDistribution, (mean, std))
 
+
+#-------------------------------------------------------------------------------
+# Inverse Normal distribution ----------------------------------------------------------
+
+
+class NormalInverseDistribution(SingleContinuousDistribution):
+    _argnames = ('mean', 'shape')
+
+    @property
+    def set(self):
+        return Interval(0, oo)
+
+    @staticmethod
+    def check(mean, shape):
+        _value_check(shape > 0, "Shape parameter must be positive")
+        _value_check(mean > 0, "Mean must be positive")
+
+    def pdf(self, x):
+        return exp(-self.shape*(x - self.mean)**2 / (2*x*self.mean**2)) * sqrt(self.shape/((2*pi*x**3)))
+
+    def sample(self):
+        scipy = import_module('scipy')
+        if scipy:
+            from scipy.stats import invgauss
+            return invgauss.rvs(float(self.mean/self.shape), 0, float(self.shape))
+        else:
+            raise NotImplementedError('Sampling the Inverse Normal Distribution requires Scipy.')
+
+    def _standardNormal_cdf(self, x):
+        return erf(sqrt(2)*(x - S.One)/2)/2 + S.Half
+
+    def _cdf(self, x):
+        mean, shape = self.mean, self.shape
+        return self._standardNormal_cdf(sqrt(shape/x) * ((x/mean - S.One))) + exp(2*shape/mean)*\
+            self._standardNormal_cdf(-sqrt(shape/x)*(x/mean + S.One))
+
+    def _characteristic_function(self, t):
+        mean, shape = self.mean, self.shape
+        return exp((shape/mean)*(1 - sqrt(1 - (2*mean**2*I*t)/shape)))
+
+    def _moment_generating_function(self, t):
+        mean, shape = self.mean, self.shape
+        return exp((shape/mean)*(1 - sqrt(1 - (2*mean**2*t)/shape)))
+
+
+def NormalInverse(name, mean, shape):
+    r"""
+    Create a continuous random variable with an Inverse Normal distribution.
+
+    The density of the Inverse Normal distribution is given by
+
+    .. math::
+        f(x) := \sqrt(\frac{\lambda}{2\pi x^3}) e^{-\frac{\lambda(x-\mu)^2}{2x\mu^2}}
+
+    Parameters
+    ==========
+
+    mu : Positive number representing the mean
+    lambda : Positive number representing the shape parameter,
+
+    Returns
+    =======
+
+    A RandomSymbol.
+
+    Examples
+    ========
+
+    >>> from sympy.stats import NormalInverse, density, E, std, cdf, skewness
+    >>> from sympy import Symbol, simplify, factor, together, factor_terms
+
+    >>> mu = Symbol("mu", positive=True)
+    >>> l = Symbol("lambda", positive=True)
+    >>> z = Symbol("z")
```

Comment:
Will it be good to add the assumption, `positive=True`?

Reply:
Done